### PR TITLE
Adding nxos_portchannel

### PR DESCRIPTION
**Pull request by GGabriele**
##### ISSUE TYPE
- New Module Pull Request
##### COMPONENT NAME

nxos_portchannel
##### ANSIBLE VERSION

```
ansible 2.2.0
  config file = /etc/ansible/ansible.cfg
  configured module search path = ['/usr/share/ansible/']
```
##### SUMMARY

Manages port-channel interfaces

### CODE REVIEW

**Comment by gregdek** [comment]
Thanks again to @GGabriele. This module is going into community review.

We notice that you are one of the maintainers (@privateip, @GGabriele, @gundalow) of this module, so when you think it's ready, please comment "shipit" and we will consider it for merging.

[This message brought to you by your friendly Ansibull-bot.]

File: network/nxos/nxos_portchannel.py [hunk 1]
```diff
@@ -0,0 +1,742 @@
...
+    for command in command_lists:
+        if isinstance(command, list):
+            flat_command_list.extend(command)
+        else:
+            flat_command_list.append(command)
+    return flat_command_list
+
+
+def main():
+    argument_spec = dict(
```
Comment (gundalow) [inline network/nxos/nxos_portchannel.py:631]: There are more options defined here than in the `DOCUMENTATION:`

Any option specified in `extends_documentation_fragment: nxos` needs defining in `module_utils`, have a look at

**Comment by gundalow** [comment]
needs_revision

**Comment by GGabriele** [comment]
@gundalow can you articulate a little bit more? I'm not getting what changes I should do in regards to `extends_documentation_fragment`.

**Comment by gundalow** [comment]
Sure, that wasn't that clear.

For any option you have defined in `extends_documentation_fragment` you should have to define that in every module in `argument_spec`, instead define it in 

See `ansible/lib/ansible/module_utils/f5.py` `f5_argument_spec()`

Then in your module you can do something similar to `ansible-modules-extras/network/f5/bigip_device_dns.py`

**Comment by gregdek** [comment]
Thanks @GGabriele for this PR. This PR requires revisions, either because it fails to build or by reviewer request. Please make the suggested revisions. When you are done, please comment with text 'ready_for_review' and we will put this PR back into review.

[This message brought to you by your friendly Ansibull-bot.]

**Pull request merged**